experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -0,0 +1,540 @@
+import ClimaParams
+import ClimaLand as CL
+import Dates
+import ClimaUtilities.TimeVaryingInputs: LinearInterpolation, PeriodicCalendar, TimeVaryingInput
+import ClimaCoupler: Checkpointer, FieldExchanger, FluxCalculator, Interfacer, Utilities
+import ClimaCore as CC
+import SciMLBase
+import ClimaTimeSteppers as CTS
+import ClimaDiagnostics as CD
+import ClimaUtilities.TimeManager: ITime
+
+include("climaland_helpers.jl")
+
+"""
+    ClimaLandSimulation{M, P, D, I, A}
+
+The integrated ClimaLand model simulation object.
```

Comment:
Perhaps we could indicate what the types of these fields are in the doc string

Reply:
We may also wish to place restrictions? e.g. M <: ClimaLand.LandModel

Reply:
and we could enforce D <: ClimaLand.Domains.SphericalShell instead of the @assert below?

Reply:
Yes good points! I'll change these here and for the bucket. We can remove the `domain_type` argument altogether with this change too. It's unused right now and if we want to support running on other domains in the future it should be specified for the overall system rather than just for land via `land_domain_type` like we have right now.